enterprise/api/v1/views.py

```diff
@@ -1249,3 +1250,68 @@ def post(self, request):
                 {'error': str('Notification read request failed')},
                 status=HTTP_500_INTERNAL_SERVER_ERROR
             )
+
+
+class EnterpriseCustomerReportTypesView(APIView):
+    """
+    API for getting the report types associated with an enterprise customer
+    """
+    authentication_classes = [SessionAuthentication]
+    permission_classes = [permissions.IsAuthenticated]
+    http_method_names = ['get']
+
+    @staticmethod
+    def _get_data_types_with_recent_progress_type(data_types):
+        """
+        Get the data types with only the most recent 'progress' type version
+
+        :param data_types: List of data type tuples
+        :return: List of data type tuples with only the most recent 'progress' type. [ ... ('progress', 'progress_v3')]
+        """
+        progress_data_types = [data_type for data_type in data_types if data_type[1].startswith('progress')]
+        progress_data_types.sort(key=lambda data_type: data_type[1])
+        data_types_for_frontend = [data_type for data_type in data_types if 'progress' not in data_type[1]]
```

Comment:
I this It would be better to use opposite of the above condition here.
`data_types_for_frontend = [data_type for data_type in data_types if not data_type[1].startswith('progress')]`